canteenWeb/views.py

```diff
@@ -67,3 +67,45 @@ def reject(self, request, pk=None):
         order.status = choices.STATUS_DICTIONARY["Rejected by Canteen"]
         order.save()
         return Response({"message": "Order rejected"})
+
+    @action(detail=False)
+    def completed(self, request):
+        completed_orders = Order.objects.filter(is_fulfilled=True)
+        serializer = self.get_serializer(completed_orders, many=True)
+        return Response(serializer.data)
+
+    @action(detail=False)
+    def pending(self, request):
+        pending_orders = Order.objects.filter(
+            is_fulfilled=False, status__gte=0
+        )  # Should not be fulfilled and status should be positive.
+        serializer = self.get_serializer(pending_orders, many=True)
+        return Response(serializer.data)
+
+    @action(detail=False)
+    def status_options(self, request):
+        return Response(choices.STATUS_DICTIONARY)
+
+    @action(detail=True, methods=["get", "post"])
+    def change_status(self, request, pk=None):
+        order = self.get_object()
+        data = request.data
+        if "status" in data:
+            if "status" in choices.STATUS_DICTIONARY_REVERSE:
```

Comment:
Won't the key of the reverse dictionary be int(data["status"]), rather than "status" itself?

Reply:
Nice catch... I didn't test this commit: https://github.com/vixrant/canteen-automation-web/commit/4525724d9ceb24350edabecebcf0fb1793b0b731

Reply:
Fixed in new commit 👍🏻